README_INTERFACE_TESTS.md

```diff
@@ -264,20 +267,42 @@ If it says `NOT OK`, there is an error in the schema format or the filename.
 
 ### Referencing the schema in an interface test
 When you write an interface test for `ingress`, by default, the test case will validate the relation against the schema provider in `schema.py` (using the appropriate role).
+In more complex cases, e.g. if the schema can assume one of multiple shapes depending on the position in a sequence of data exchanges, it can be handy to override that default.
 
-`interface_tester.interface_test_case` accepts a `schema` argument that allows you to configure this behaviour. It can take one of four possible values:
-- `interface_tester.interface_test.SchemaConfig.default` (or the string `"default"`): validate any `ingress` relation found in the `state_out` against the schema found in `schema.py`. If this interface test case is for the requirer, it will use the `RequirerSchema`; otherwise the `ProviderSchema`.
-- `interface_tester.interface_test.SchemaConfig.skip` (or the string`"skip"`): skip schema validation for this test.
-- `interface_tester.interface_test.SchemaConfig.empty` (or the string`"empty"`): assert that any `ingress` relation found in the `state_out` has **no relation data** at all (local side).
-- you can pass a custom `interface_tester.schema_base.DataBagSchema` subclass, which will be used to validate any `ingress` relation found in `state_out`. This will replace the default one found in `schema.py` for this test only.
-
+`Tester.assert_schema_valid` accepts a `schema` argument that allows you to configure the expected schema.
+Pass to it a custom `interface_tester.DataBagSchema` subclass and that will replace the default schema for this test.
 
 # Matrix-testing interface compliance
 If we have:
 - a `../interfaces/ingress/v0/charms.yaml` listing some providers and some requirers of the `ingress` interface.
 - a `../interfaces/ingress/v0/schema.py` specifying the interface schema (optional: schema validation will be skipped if not found)
-- a `../interfaces/ingress/v0/interface_tests/my_tests.py` providing a list of interface tests for either role
+- two `../interfaces/ingress/v0/interface_tests/test_[requirer|provider].py` files providing a list of interface tests for either role
 
 You can then run `python ./run_matrix.py ingress`.
 This will attempt to run the interface tests on all charms in `.../interfaces/ingress/v0/charms.yaml`.
 Omitting the `ingress` argument will run the tests for all interfaces (warning: might take some time.)
+
+# Charm repo configuration
+When developing the tests, it can be useful to run them against a specific branch of a charm repo. To do that, write in `charms.yaml`:
+
+```yaml
+providers:
+  - name: traefik-k8s
+    url: https://github.com/canonical/traefik-k8s-operator
+    branch: develop  # any custom branch
+
+requirers: []
```

Comment:
Here we introduce a yaml configuration file. I think it may be worth it to better document it. I'd recommend to look at [Pebble's Layer specification](https://github.com/canonical/pebble#layer-specification) and do something similar in the docs here.

Reply:
added a separate doc for that